packages/env/lib/detect-context.js

```diff
@@ -13,21 +13,45 @@ const path = require( 'path' );
 const readDir = util.promisify( fs.readdir );
 const finished = util.promisify( stream.finished );
 
-module.exports = async function detectContext() {
+const cwd = process.cwd();
+
+/**
+ * @typedef Context
+ * @type {Object}
+ * @property {string} type
+ * @property {string} path
+ * @property {string} pathName
+ */
+
+/**
+ * Detects the context of a given path.
+ *
+ * @param {string} directoryPath The directory to detect. Should point to a directory, defaulting to the current directory.
```

Comment:
We need to document default parameters.

https://jsdoc.app/tags-param.html#optional-parameters-and-default-values